### ISSUE TITLE
Issue with summarise()/count() with dbplyr v. 2.2.1

### ISSUE THREAD

**Issue by andreassoteriadesmoj** [tidyverse/dbplyr#969]
Hi, 

The blow code works fine with `dbplyr` v. 2.1.1.9000 for a `tbl` object named `re`:

```
groups <- 'event_id'
re %>%
    dplyr::distinct() %>%
    dplyr::group_by(
      dplyr::across(
        dplyr::all_of(groups)
      )
    ) %>%
    dplyr::summarize(n = dplyr::n())
```

However, running the above with `dbplyr` v. 2.2.1, the following error occurs:

```
Error in `.f()`:
! Can't convert NULL to a string.
Run `rlang::last_error()` to see where the error occurred.

rlang::last_error()

17: stop(fallback)
16: signal_abort(cnd, .file)
15: abort(msg, call = call)
14: abort_coercion(x, "a string")
13: .f(.x[[i]], ...)
12: purrr::map_chr(vars, as_string)
11: update_lazy_select(select, vars)
10: add_select(cur_data, set_names(list(NULL), dot_name), "select")
9: summarise_bind_error(cur_data, dots, i, error_env)
8: summarise_eval_dots(.data, ...)
7: summarise.tbl_lazy(x, `:=`(!!name, !!n))
6: summarise(x, `:=`(!!name, !!n))
5: tally.tbl_sql(out, wt = !!enquo(wt), sort = sort, name = name)
4: tally(out, wt = !!enquo(wt), sort = sort, name = name)
3: count.tbl_sql(., event_id)
2: dplyr::count(., event_id)
1: re %>% dplyr::ungroup() %>% dplyr::count(event_id)
```

Thanks.

**Comment by andreassoteriadesmoj** [comment]
Same issue if `count()` is used instead of `summarise()`.

**Comment by hadley** [comment]
Can you please make a reprex, following the advice at https://dbplyr.tidyverse.org/articles/reprex.html ?

**Comment by andreassoteriadesmoj** [comment]
This is weird. A reprex with toy data is working just fine with v. 2.2.1:

```
library(magrittr)

re <- dbplyr::memdb_frame(event_id = c(1, 1, 2, 2, 2), x = 1:5, y = 5:1)

groups <- 'event_id'
re %>%
  dplyr::distinct() %>%
  dplyr::group_by(
    dplyr::across(
      dplyr::all_of(groups)
    )
  ) %>%
  dplyr::summarize(n = dplyr::n())


# Source:   SQL [2 x 2]
# Database: sqlite 3.39.1 [:memory:]
  event_id     n
     <dbl> <int>
1        1     2
2        2     3
```

```
re %>%
  dplyr::distinct() %>%
  dplyr::group_by(
    dplyr::across(
      dplyr::all_of(groups)
    )
  ) %>%
 dplyr::count()

# Source:   SQL [2 x 2]
# Database: sqlite 3.39.1 [:memory:]
  event_id     n
     <dbl> <int>
1        1     2
2        2     3
```

I don't have a good reason- other than ignorance- for asking this, but do you think that the reason behind the error may be that my own database object is of class `tbl_MoJAthenaConnection`?

**Comment by hadley** [comment]
Yes, it might be specific to MoJAthenaConnection, in which case you should contact the maintainer of whatever package produde it.

**Comment by mgirlich** [comment]
@andreassoteriadesmoj The issue might have been fixed by #988. You might want to install the dev version and check.
Closing this issue now as this cannot be reproduced and might be related to MoJAthenaConnection.
